src/pepr/operator/controllers/secrets/pod-reload.ts

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright 2025 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+import { createHash } from "crypto";
+import { K8s, kind } from "pepr";
+import { Component, setupLogger } from "../../../logger";
+import { reloadPods } from "../utils";
+
+const log = setupLogger(Component.OPERATOR_SECRETS);
+
+// Map to store secret checksums for change detection
+// Exported for testing purposes
+export const secretChecksumCache = new Map<string, string>();
+
+/**
+ * Computes a SHA256 checksum of the secret data
+ *
+ * @param data The secret data to compute the checksum for
+ * @returns A hex-encoded SHA256 checksum
+ */
+export function computeSecretChecksum(data: Record<string, string>): string {
+  // Sort keys to ensure consistent hashing regardless of key order
+  const sortedKeys = Object.keys(data).sort();
+  const hash = createHash("sha256");
+
+  // Add each key-value pair to the hash
+  for (const key of sortedKeys) {
+    hash.update(`${key}=${data[key]}`);
+  }
+
+  return hash.digest("hex");
+}
+
+/**
+ * Auto-discovers pods that use the given secret
+ *
+ * @param namespace Namespace of the secret
+ * @param secretName Name of the secret
+ * @returns Array of pods that mount or reference the secret
+ */
+async function discoverSecretConsumers(namespace: string, secretName: string) {
+  // Get all pods in the namespace
+  const pods = await K8s(kind.Pod).InNamespace(namespace).Get();
+
+  // Filter pods that use the secret either as a volume or env var source
+  return pods.items.filter(pod => {
+    if (!pod.spec) return false;
+
+    // Check volume mounts for direct secret volumes
+    const usesSecretVolume = pod.spec.volumes?.some(
+      volume => volume.secret && volume.secret.secretName === secretName,
+    );
+    if (usesSecretVolume) return true;
+
+    // Check for projected volumes that include the secret
+    const usesProjectedSecretVolume = pod.spec.volumes?.some(volume => {
+      if (!volume.projected || !volume.projected.sources) return false;
+
+      return volume.projected.sources.some(
+        source => source.secret && source.secret.name === secretName,
+      );
+    });
```

Comment:
nit: this is me trying to understand my TS syntax. I think this would be cleaner?
```suggestion
    const usesProjectedSecretVolume = pod.spec.volumes?.some(
      volume =>
        volume.projected?.sources?.some(
          source => source.secret?.name === secretName,
      ),
);
```

Reply:
Agreed - this aligns closer with the approach for env as well. Will update.